### DISCUSSION TITLE
How to setup YourTTS model for transfer learning with a new language

### DISCUSSION THREAD

**Discussion by dostee-hemin** [coqui-ai/TTS#1957] [General Q&A]
I recently came across the [YourTTS ](https://edresson.github.io/YourTTS/) model, and was wondering how it could be trained using transfer learning from the provided [pretrained models](https://github.com/Edresson/YourTTS). My end goal is to train this model on a language it has never seen before, but hopefully by continuing training from a pretrained model, I shouldn't need that much time or data.

From my understanding, I should be able to go into the config.json file and add the necessary characters, punctuation, dataset, ... etc. of my new language and continue the training process with the new configurations so it starts learning the new language. At my current state, I have a dataset ready, but the following are still unclear:
1) In what format should I organize my metadata.csv file? LJSpeech or something else?
2) I haven't found any train.py files online that are used for the YourTTS model; what exactly should I write in the train.py file?
3) What are the settings in the config.json file I need to change so that the model can accept this new language?
4) I have come across terms such as speaker embeddings and their respective config.json and model paths, what do they mean and how exactly do I fit them in the training and inference processes?

Any insight is greatly appreciated.

**Comment by jeffersonquispe** [comment]
The answer to this would be a great contribution

**Comment by Ca-ressemble-a-du-fake** [comment]
@dostee-hemin regarding your questions : 
1. You can use LJSpeech format it is very simple (look at tts/datasets/formatters.py and look for ljspeech function to understand what is expected)
2. You can use train-tts.py which lies in TTS.TTS/bin and provide a config.json
3. For French I started over from scratch, it was much easier and not that long until VITS model gave amazing results (although not perfect but enough for me).
4. Speaker embeddings are for multispeaker model I gather. So for single speaker training you don't need to define them manually.

**Comment by erogol** [comment]
Have you checked https://tts.readthedocs.io/?
